File: conformance/runners/swift/runner.swift
// SPDX-License-Identifier: Apache-2.0
// SPDX-FileCopyrightText: 2026 Nirapod Labs

// Signet conformance runner (Swift), scaffold stub.
// Answers every behavior with "unimplemented", one answer per request.

import Foundation

while let line = readLine() {
    let text = line.trimmingCharacters(in: .whitespacesAndNewlines)
    if text.isEmpty { continue }
    guard let data = text.data(using: .utf8),
          let object = try? JSONSerialization.jsonObject(with: data) as? [String: Any],
          let behavior = object["behavior"] as? String else { continue }
    print("{\"behavior\":\"\(behavior)\",\"status\":\"unimplemented\"}")
}

File: conformance/runners/typescript/runner.mjs
// SPDX-License-Identifier: Apache-2.0
// SPDX-FileCopyrightText: 2026 Nirapod Labs

// Signet conformance runner (TypeScript/Node), scaffold stub.
// Answers every behavior with "unimplemented", one answer per request; an
// unimplemented behavior is never a silent skip. Real assertions replace the
// stub as the binding lands.

import { createInterface } from 'node:readline'

const lines = createInterface({ input: process.stdin, terminal: false })
lines.on('line', (line) => {
  const text = line.trim()
  if (!text) return
  let request
  try {
    request = JSON.parse(text)
  } catch {
    return
  }
  if (!request.behavior) return
  process.stdout.write(
    `${JSON.stringify({ behavior: request.behavior, status: 'unimplemented' })}\n`,
  )
})
